src/test/java/main/Tasks.java

```diff
@@ -0,0 +1,83 @@
+package main;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+import entities.BinarySearch;
+import entities.Data;
+import entities.Entity;
+import entities.MergeSort;
+import entities.Searcher;
+import entities.Sorter;
+
+public class Tasks {
+
+    @Test
+    public void test_2_3_7() {
+        List<Entity> list = new ArrayList<>();
+        
+        list.add(new Entity(Data.getNextName(), 50));
+        list.add(new Entity(Data.getNextName(), 48));
+        list.add(new Entity(Data.getNextName(), 47));
+        list.add(new Entity(Data.getNextName(), 46));
+        list.add(new Entity(Data.getNextName(), 45));
+        list.add(new Entity(Data.getNextName(), 40));
+        list.add(new Entity(Data.getNextName(), 25));
+        list.add(new Entity(Data.getNextName(), 22));
+        list.add(new Entity(Data.getNextName(), 21));
+        list.add(new Entity(Data.getNextName(), 18));
+        
+        Entity entity = new Entity(Data.getNextName(), 96);
+        
+        test2_3_7(list, entity, 6, 9);
+        
+    }
+
+    /*
+     * Задача 2.3.7 Разработайте алгоритм со вмеренем работы n*lg(n) (где lg это
```

Comment:
Going further, I would suggest to use entity with some attribute that can be negative (instead of `age` - `ratingScore` for example). So that your assumptions about content of the incoming data won't affect your solution.